tensorflow/compiler/tf2tensorrt/convert/ops/tile.cc

```diff
@@ -0,0 +1,169 @@
+    }
+
+    nvinfer1::DataType dtype;
+    const int *multiplies;
+    if (repl.is_weights()) {
+      CHECK_SHAPE_TENSOR(repl.weights().GetTensor());
+      dtype = repl.weights().TrtDType();
+      multiplies = repl.weights().GetPointer<int>();
+    } else {
+      dtype = repl.tensor()->getType();
+      multiplies = nullptr;
+    }
+
+    if (dtype != nvinfer1::DataType::kINT32) {
+      return errors::InvalidArgument("The replication parameter of the ", params.node_def.op(), " operation in ",
+                                     params.node_def.name(), " is expected to be of ",
+                                     DebugString(nvinfer1::DataType::kINT32),
+                                     " type, got ", DebugString(dtype));
+    }
+
+    const auto dims = inputs.at(0).GetTrtDims();
+    const auto nb_dims = dims.nbDims + (params.use_implicit_batch && inputs.at(0).is_tensor()? 1 : 0);
+    if (multiplies) {
+      const int mult_numb = repl.weights().count();
+      if (mult_numb != nb_dims) {
+        return errors::InvalidArgument("The lenght of the replication vector (", mult_numb,
+                                       ") of the Tile operation in '", params.node_def.name(),
+                                       "' is expected to be equal to the number of dimentions of the input vector (",
+                                       nb_dims, ")");
+      }
+
+      bool positiveReplic = true;
+      for (int i = 0; i < nb_dims; i++)
+        positiveReplic &= multiplies[i] > 0;
+
+      if (!positiveReplic) {
+        char buffer[256], *pBuff = buffer;
+        for (int i = 0; i < nb_dims; i++)
+          pBuff += sprintf(pBuff, (i? ", %d" : "%d"), multiplies[i]);
+
+         return errors::InvalidArgument("All replications of the Tile operation in ", params.node_def.name(),
+                                     " should be positive, got (", buffer, ").");
+      }
+    }
+
+    return Status::OK();
+  }
+
+  Status Convert() {
+    const auto &params = *this->params_;
+    const auto& inputs = params.inputs;
+    auto *converter = params.converter;
+    auto *network = converter->network();
+    const auto &tensor = inputs.at(0);
+    const auto &replics = inputs.at(1);
+    const auto dims = tensor.GetTrtDims();
+    const auto nb_dims = dims.nbDims;
+    const auto dim_adj = params.use_implicit_batch && tensor.is_tensor()? 1 : 0;
+
+    bool dynamic_flag = false;
+    const int *pMultiplies = nullptr;
+    if (replics.is_tensor()) {
+      dynamic_flag = true;
+    } else {
+      pMultiplies = replics.weights().GetPointer<int>();
+    }
+
+    nvinfer1::Dims size{nb_dims, {1}};
+    nvinfer1::Dims stride{nb_dims, {1}};
+    const auto *pSize = dims.d;
+    for (int i = 1 - dim_adj; i < nb_dims; i++) {
+      stride.d[i] = 1;
+      if (pMultiplies)
+        dynamic_flag |= (size.d[i] = pMultiplies[dim_adj+i] * *(pSize+i)) < 0;
+    }
+
```

Comment:
For manipulating `INetworkDefinition*` in below code, consider using `NetworkBuilder` in `ops/layer_utils.h`. This provides better encapsulation of network manipulation across the code base.

Reply:
Do you mean `TRTNetworkBuilder`?

Reply:
yes

Reply:
Implemented the use of TRTNetworkBuilder